pandas/core/strings.py

```diff
@@ -1095,24 +1095,61 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
-    Split each string (a la re.split) in the Series/Index by given
-    pattern, propagating NA values. Equivalent to :meth:`str.split`.
+    Split strings around given separator/delimiter.
+
+    Split each string in the Series/Index by given
+    pattern, propagating NaN values. Equivalent to :meth:`str.split`.
 
     Parameters
     ----------
     pat : string, default None
-        String or regular expression to split on. If None, splits on whitespace
+        String or regular expression to split on.\
+         If `None`, split on whitespace.
     n : int, default -1 (all)
-        None, 0 and -1 will be interpreted as return all splits
+        Vary dimensionality of output.
+
+        * `None`, 0 and -1 will be interpreted as return all splits
     expand : bool, default False
-        * If True, return DataFrame/MultiIndex expanding dimensionality.
-        * If False, return Series/Index.
+        Expand the split strings into separate columns.
 
-    return_type : deprecated, use `expand`
+        * If `True`, return DataFrame/MultiIndex expanding dimensionality.
+        * If `False`, return Series/Index.
 
     Returns
     -------
+    Type matches caller unless `expand=True` (return type is `DataFrame`)
     split : Series/Index or DataFrame/MultiIndex of objects
+
+    Notes
+    -----
+    If `expand` parameter is `True` and:
+      - If n >= default splits, makes all splits
+      - If n < default splits, makes first n splits only
+      - Appends `None` for padding.
+
+    Examples
```

Comment:
Might have missed mentioning this but while I think the examples are good you should add a sentence (or a few) to call out what users should be looking at with the examples. It would also be nice to show one or two with missing data to illustrate how `NaN` gets propagated

Reply:
Like this?
```
>>> s.str.split("is", n=1, expand=True)
        0                1
0      th     is good text
1  but th   is even better
see notes about expand=True
```
Also, I wrote an example to show NaN propagation but None is being propagated instead of NaN.
```
>>> s = pd.Series(["this is good text", "but this is even better", np.nan])
>>> s.str.split(expand=True)
      0     1     2     3       4
0  this    is  good  text    None
1   but  this    is  even  better
2   NaN  None  None  None    None
```

Reply:
Put your comments before the example and just highlight what the user should look at. So for your first example say something like "By default, split will return an object of the same size containing lists to hold the split elements" and then introduce the second with something like "By contrast, when using `expand=True` the split elements will expand out into separate columns." Doesn't need to be exactly those words but something along those lines - make sense?

As far as your example is concerned, make sure you run everything on the master branch. My guess is you are using an older version of pandas as the fix to propagate `NaN` was released in v0.21.1 (see https://github.com/pandas-dev/pandas/pull/18462)

Reply:
ah, right.
thanks!